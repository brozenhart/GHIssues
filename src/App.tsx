import { NavigationContainer } from '@react-navigation/native';
import React from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { IssuesStackNavigator, NavigationRouteName } from '@/navigators';
import { Strings } from '@/config';

const Tab = createBottomTabNavigator();

export const App = (): JSX.Element => {
  return (
    <NavigationContainer>
      <Tab.Navigator initialRouteName={NavigationRouteName.ISSUES}>
        <Tab.Screen
          name={NavigationRouteName.ISSUES}
          component={IssuesStackNavigator}
          options={{
            tabBarLabel: Strings.TAB_BAR_LABEL_ISSUES,
          }}
        />
      </Tab.Navigator>
    </NavigationContainer>
  );
};
