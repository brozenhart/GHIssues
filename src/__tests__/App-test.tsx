import { render } from '@testing-library/react-native';
import 'react-native';
import React from 'react';
import App from '../App';

describe('App', () => {
  it('should render correctly', () => {
    const page = render(<App />);
    expect(page.toJSON()).toMatchSnapshot();
  });
});
