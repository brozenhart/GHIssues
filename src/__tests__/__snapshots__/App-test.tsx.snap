// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenContainer
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RNSScreen
      activityState={2}
      forwardedRef={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "zIndex": 1,
              }
            }
          >
            <View
              accessibilityElementsHidden={false}
              importantForAccessibility="auto"
              onLayout={[Function]}
              pointerEvents="box-none"
              style={null}
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "zIndex": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "rgb(255, 255, 255)",
                      "borderBottomColor": "rgb(216, 216, 216)",
                      "flex": 1,
                      "shadowColor": "rgb(216, 216, 216)",
                      "shadowOffset": Object {
                        "height": 0.5,
                        "width": 0,
                      },
                      "shadowOpacity": 0.85,
                      "shadowRadius": 0,
                    }
                  }
                />
              </View>
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "height": 64,
                    "maxHeight": undefined,
                    "minHeight": undefined,
                    "opacity": undefined,
                    "transform": undefined,
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "height": 20,
                    }
                  }
                />
                <View
                  pointerEvents="box-none"
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                >
                  <View
                    pointerEvents="box-none"
                    style={
                      Object {
                        "marginHorizontal": 16,
                        "opacity": 1,
                      }
                    }
                  >
                    <Text
                      accessibilityRole="header"
                      aria-level="1"
                      numberOfLines={1}
                      onLayout={[Function]}
                      style={
                        Object {
                          "color": "rgb(28, 28, 30)",
                          "fontSize": 17,
                          "fontWeight": "600",
                        }
                      }
                    >
                      Repositories
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <RNSScreenContainer
            onLayout={[Function]}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <RNSScreen
              activityState={2}
              forwardedRef={[Function]}
              pointerEvents="box-none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              />
              <View
                accessibilityElementsHidden={false}
                closing={false}
                gestureVelocityImpact={0.3}
                importantForAccessibility="auto"
                onClose={[Function]}
                onGestureBegin={[Function]}
                onGestureCanceled={[Function]}
                onGestureEnd={[Function]}
                onOpen={[Function]}
                onTransition={[Function]}
                pointerEvents="box-none"
                style={
                  Array [
                    Object {
                      "overflow": undefined,
                    },
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                  ]
                }
                transitionSpec={
                  Object {
                    "close": Object {
                      "animation": "spring",
                      "config": Object {
                        "damping": 500,
                        "mass": 3,
                        "overshootClamping": true,
                        "restDisplacementThreshold": 10,
                        "restSpeedThreshold": 10,
                        "stiffness": 1000,
                      },
                    },
                    "open": Object {
                      "animation": "spring",
                      "config": Object {
                        "damping": 500,
                        "mass": 3,
                        "overshootClamping": true,
                        "restDisplacementThreshold": 10,
                        "restSpeedThreshold": 10,
                        "stiffness": 1000,
                      },
                    },
                  }
                }
              >
                <View
                  needsOffscreenAlphaCompositing={false}
                  pointerEvents="box-none"
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    forwardedRef={[Function]}
                    onGestureHandlerEvent={[Function]}
                    onGestureHandlerStateChange={[Function]}
                    style={
                      Object {
                        "flex": 1,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateX": 0,
                          },
                        ],
                      }
                    }
                  >
                    <View
                      pointerEvents="none"
                      style={
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "shadowColor": "#000",
                          "shadowOffset": Object {
                            "height": 1,
                            "width": -1,
                          },
                          "shadowOpacity": 0.3,
                          "shadowRadius": 5,
                          "top": 0,
                          "width": 3,
                        }
                      }
                    />
                    <View
                      style={
                        Array [
                          Object {
                            "flex": 1,
                            "overflow": "hidden",
                          },
                          Array [
                            Object {
                              "backgroundColor": "rgb(242, 242, 242)",
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                            "flexDirection": "column-reverse",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "flex": 1,
                                "justifyContent": "flex-start",
                                "paddingTop": 20,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "backgroundColor": "#FFFFFF",
                                  "flex": 0,
                                  "flexDirection": "row",
                                  "marginBottom": 10,
                                  "paddingHorizontal": 10,
                                  "paddingVertical": 20,
                                }
                              }
                            >
                              <Text
                                style={
                                  Object {
                                    "flex": 0,
                                  }
                                }
                              >
                                Github Organisation:
                              </Text>
                              <TextInput
                                allowFontScaling={true}
                                autoCapitalize="none"
                                blurOnSubmit={true}
                                collapsable={false}
                                onGestureHandlerEvent={[Function]}
                                onGestureHandlerStateChange={[Function]}
                                onSubmitEditing={[Function]}
                                placeholder="facebook, apple..."
                                rejectResponderTermination={true}
                                returnKeyType="search"
                                style={
                                  Object {
                                    "backgroundColor": "#FFFFFF",
                                    "flex": 1,
                                    "marginLeft": 10,
                                  }
                                }
                                underlineColorAndroid="transparent"
                              />
                            </View>
                            <RCTScrollView
                              ListEmptyComponent={[Function]}
                              collapsable={false}
                              data={Array []}
                              disableVirtualization={false}
                              getItem={[Function]}
                              getItemCount={[Function]}
                              horizontal={false}
                              initialNumToRender={10}
                              keyExtractor={[Function]}
                              maxToRenderPerBatch={10}
                              onContentSizeChange={[Function]}
                              onEndReachedThreshold={2}
                              onGestureHandlerEvent={[Function]}
                              onGestureHandlerStateChange={[Function]}
                              onLayout={[Function]}
                              onMomentumScrollEnd={[Function]}
                              onScroll={[Function]}
                              onScrollBeginDrag={[Function]}
                              onScrollEndDrag={[Function]}
                              removeClippedSubviews={false}
                              renderItem={[Function]}
                              renderScrollComponent={[Function]}
                              scrollEventThrottle={50}
                              stickyHeaderIndices={Array []}
                              updateCellsBatchingPeriod={50}
                              viewabilityConfigCallbackPairs={Array []}
                              windowSize={21}
                            >
                              <View>
                                <View
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "flex": 1,
                                      "flexDirection": "column",
                                      "marginTop": 30,
                                    }
                                  }
                                >
                                  <Text
                                    style={
                                      Object {
                                        "color": "#696969",
                                      }
                                    }
                                  >
                                    No repositories found
                                  </Text>
                                </View>
                              </View>
                            </RCTScrollView>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </RNSScreen>
          </RNSScreenContainer>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenContainer>
  <View
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "rgb(255, 255, 255)",
        "borderTopColor": "rgb(216, 216, 216)",
        "borderTopWidth": 0.5,
        "bottom": 0,
        "elevation": 8,
        "height": 49,
        "left": 0,
        "paddingBottom": 0,
        "paddingHorizontal": 0,
        "position": null,
        "right": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="Issues, tab, 1 of 1"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": true,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              Object {
                "color": "rgb(0, 122, 255)",
              },
              Object {
                "fontSize": 10,
              },
              undefined,
            ]
          }
        >
          Issues
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
